src/index.js

```diff
@@ -1,7 +1,349 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React from "react";
+import ReactDOM from "react-dom";
+import VTMap from "./Components/VTMap.js";
+import Button from "./Components/Buttons.js";
+import Display from "./Components/InfoBox.js";
+import borderData from "./Components/border.js";
+import leafletPip from "leaflet-pip";
+import L from "leaflet";
+import GuessBox from "./Components/guessBox.js"
+import "./CSS/index.css" 
+import Move from "./Components/Move.js"
+import GameWon from "./Components/gameWon.js"
 
-ReactDOM.render(
-  <h1>Hello World!</h1>,
-  document.getElementById('root')
-)
\ No newline at end of file
+let localStore = window.localStorage;
+
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+
+    this.state = {
+      initialLat: 43.9,
+      initialLng: -72.7317,
+      currentLat: 43.9,
+      currentLng: -72.7317,
+      town: "?",
+      county: "?",
+      latDisplay: "?",
+      lngDisplay: "?",
+      startDisabled: false,
+      guessDisabled: true,
+      quitDisabled: true,
+      zoom: 8,
+      modal: false,
+      winModal: false,
+      score: 100,
+      moveArr: [],
+      allScores: localStore.getItem("scores") || "",
+      highScore: "",
+      userName: ""
+    };
+  }
+
+//Getting all scores
+componentDidMount(){
+  let allScores= this.state.allScores;
+  JSON.parse(allScores); 
+  let highestScore= JSON.parse(allScores).reduce((previousValue, nextValue)=>{
+    if (parseInt(previousValue.score) > parseInt (nextValue.score)){
+      return previousValue
+    }else{
+      return nextValue
+    }
+  });
+  console.log(highestScore)
+  this.setState({
+    highScore: highestScore.score + ' ' + highestScore.name
+  })
+}
```

Comment:
This currently breaks the game on the first time it's launched on a computer, It is unable to do a JSON.parse if there is no information being sent back. Putting an `if{...}` check can prevent that by making sure that there is something being saved in allScores

Reply:
Can I just change the allScores declaration to "let allScores = this.state.allScores || 0" ?

Reply:
Yes, that should work.